CONTRIBUTING.md

```diff
@@ -1,256 +1,339 @@
+This root directory name should always be in lower case and follow the official Chocolatey naming conventions: https://docs.chocolatey.org/en-us/create/create-packages#naming-your-package.
 
-### 1.2.3 Informative description
+#### Dependency versions
 
-Description is maintained in the `README.md` file in the root of the package and automatically set in the nuspec by AU during an update. It should include the following **2nd level headers (##) in the given order**:
+1. When taking a dependency on an extension, a minimum version of that dependency must be specified. Without this, any version satisfies the dependency. That means unless someone upgrades the extension outside of their process or incidentally install some package that uses newer version explictly set, it will not automatically upgrade to the latest version. Changes to a package that does not specify a minimum version, or exact version is will not be accepted.
+2. When creating a dependency for virtual packages, a exact version range should be used for the dependent package (_.install_ or _.portable), which should be the same as that of the virtual package. An exact version range in the metadata file will look like `[1.0.0, 1.0.0]`.
 
-|header                |meaning                                        |
-|---                   |---                                            |
-| `Features`           | Bullet list with basic software functionality |
-| `Package parameters` | Bullet list with of all package parameters    |
-| `Notes`              | Bullet list with any special information to the user if any |
+#### Conform to guidelines
 
-### 1.2.4 Add chocolatey-community among owners
+Conform with the [official package creation guidelines](1754-Update-Guidelines-and-code-to-reflect-new-procedures) and take a look at the [quick start guide](https://docs.chocolatey.org/en-us/create/create-packages-quick-start) on how to create packages.
 
-Keep any exisiting owners and add `chocolatey-community` user before all others.
+You should also know how to [deprecate a package](https://docs.chocolatey.org/en-us/create/create-packages-quick-start)
 
-### 1.2.5 Provide icon
+#### Obligatory metadata
 
-Packages **must have an icon** if one is available. The icon must be named the same as the package and is placed in the [icons](https://github.com/chocolatey-community/chocolatey-packages/tree/master/icons) directory.
+When possible, all elements in the nuspec file is obligated to be made use of. This includes elements like `packageSourceUrl`, `projectSourceUrl`, `docsUrl`, `bugTrackerUrl`, `releaseNotes`, `licenseUrl` and `iconUrl`, even when one or more of these are made optional on Chocolatey Community Repository itself. If the software itself does not have any sufficient URLs to be used for these examples, then it is fine to be omitted.
```

Comment:
```suggestion
When possible, all possible elements in the nuspec file should be filled. This includes elements like `packageSourceUrl`, `projectSourceUrl`, `docsUrl`, `bugTrackerUrl`, `releaseNotes`, `licenseUrl` and `iconUrl` - even when one or more of these are made optional on Chocolatey Community Repository itself. If the software itself does not have any sufficient URLs to be used for these examples then it can be omitted.
```